### DISCUSSION TITLE
What's the difference between Actor.stop, Actor.exit_after, and Actor.kill?

### DISCUSSION THREAD

**Discussion by phil-skillwon** [slawlor/ractor#161] [Q&A] [answered]
What's the difference between Actor.stop, Actor.exit_after, and Actor.kill? 
In which scenarios are they best suited for use?

**Comment by slawlor** [comment, marked as answer]
* `stop` stops the actor from processing future messages, but will finish the current message. This is a "graceful" stop
* `kill` will stop the actor at the next async await point, which can interrupt current message processing. This is a "forceful" stop
* `exit_after` will simply send the `stop` message after some delay, and can be used to schedule a stop operation. There's a similar `kill_after` which sends `kill` after some delay.

All of this should be on the docs.rs for the crate, if you find documentation lacking please let me know and I can try and expand on it where necessary.

**Reply by phil-skillwon** [reply]
Alright, thank you. I'm embarrassed; I should have read the documentation first.
